routes/bookRoute.js

```diff
@@ -1,12 +1,64 @@
 const express = require("express");
 const bookController = require("../controllers/bookController");
-
+const { body } = require("express-validator");
 const router = express.Router();
 
-router.post("/", bookController.addBook);
+router.post(
+  "/",
+  [
+    body("title").notEmpty().withMessage("A book must have a title"),
+    body("title")
+      .if(body("title").notEmpty())
+      .isLength({ min: 5 })
+      .withMessage("A book title must have more than 5 characters"),
+  ],
+  [body("genre").notEmpty().withMessage("A book must have a genre")],
+  [body("author").notEmpty().withMessage("A book must have an author")],
+  [
+    body("ISBN").notEmpty().withMessage("Please provide ISBN number"),
+    body("ISBN")
+      .if(body("ISBN").notEmpty())
+      .isLength({ min: 13, max: 13 })
+      .withMessage("ISBN number must be 13 in length"),
+  ],
+  [
+    body("overview").notEmpty().withMessage("A book must have an overview"),
+    body("overview")
+      .if(body("title").notEmpty())
+      .isLength({ min: 15 })
+      .withMessage("A book must not have an overview less than 15 characters"),
+  ],
+  bookController.addBook
+);
 router.get("/", bookController.getAllBooks);
```

Comment:
Is this endpoint paginated??? If not, this is a wrong way to implement an API. You can't serve all the books in your db to the FE at the same time. Imagine you have millions of records in your database

Reply:
In addition, let this endpoint takes care of all other possible search parameters like author, ISBN and other attributes you will like them to be able to search with. You can take them as query params on this same endpoint